### PR TITLE
Add Pypy 5.8.0

### PR DESCRIPTION
**Pull request by Samureus**
Do not merge yet, trying to follow issue https://github.com/pyenv/pyenv/issues/931

### CODE REVIEW

**Comment by Samureus** [comment]
Tested binaries and portable binaries on Ubuntu 16.04, waiting on the source translations.

**Comment by joshfriend** [comment]
Sometimes I wonder if we actually need separate build definitions for the portable PyPy versions, they get used automatically if the official linux binaries aren't supported.

**Comment by Samureus** [comment]
On the other hand, 

- we need to wait until the portable pypy binaries are ready before publishing new definitions (which is ok, approx. 12 hours delay at most)
- the official binaries seem to work fine on Ubuntu 16.04, and 12.04 is deprecated by Canonical but still being used by Travis

(still waiting on translation, old laptop + using cpython instead of pypy sucks)

**Comment by Samureus** [comment]
Yay, 2 hours translating pypy2-5.8.0-src! Gonna get me one of these new Ryzen thingies one day.

`Python 2.7.13 (c925e73810367cd960a32592dd7f728f436c125c, Jun 09 2017, 17:52:01)
[PyPy 5.8.0 with GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
And now for something completely different: ``all we need is a native Prolog
machine''
`

**Comment by yyuu** [comment]
Thanks for adding this 👍 

IIRC, my initial intention to create separate definitions for pypy-portable is that it'd be better to prepare definitions per _provider_. Though it's not so meaningful anymore since we're installing "pypy-portable" as "pypy" at least on some platforms.

I think we can just merge pypy and pypy-portable definitions (=~ stop creating pypy-portable definitions from now on). I guess not so many people are actually relying on "pypy-portable" since "pypy" is setup to install pypy-portable on the platforms where PyPy official binariy is not provided.

**Comment by Samureus** [comment]
Sorry got completely sidetracked on this. @BanzaiMan is it ok with you this way? @yyuu @joshfriend @blueyed if any of you could test this (to avoid "works on my machine" mistakes), then I guess it's ok to merge?

**Comment by BanzaiMan** [comment]
I don't think Precise can build PyPy any more. It is best to drop it. See https://github.com/pyenv/pyenv/issues/925#issuecomment-307527973.

**Comment by Samureus** [comment]
Follow up issues:

https://bitbucket.org/pypy/pypy/issues/2574/dropping-support-for-ubuntu-1204-precise
https://github.com/squeaky-pl/portable-pypy/issues/53

I'm ok for dropping Precise support myself, in fact, we could remove all of these
`"Ubuntu 12.04" "Ubuntu 12.10" "Ubuntu 13.04" "Ubuntu 13.10"` while we're at it, but Portable PyPy might not work as a substitute (at the moment), guys what do you think?

File: plugins/python-build/share/python-build/pypy2.7-5.8.0 [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  else
+    require_distro "Ubuntu 13.04" || true
+    install_package "pypy2-v5.8.0-linux-armhf-raring" "https://bitbucket.org/pypy/pypy/downloads/pypy2-v5.8.0-linux-armhf-raring.tar.bz2#ddceca9c5c9a456d4bf1beab177660adffbbdf255a922244e1cc05f20318be46" "pypy" verify_py27 ensurepip
+  fi
+  ;;
+"linux64" )
+  if require_distro "Ubuntu 14.04" 1>/dev/null 2>&1; then
+    install_package "pypy2-v5.8.0-linux64" "https://bitbucket.org/pypy/pypy/downloads/pypy2-v5.8.0-linux64.tar.bz2#6274292d0e954a2609b15978cde6efa30942ba20aa5d2acbbf1c70c0a54e9b1e" "pypy" verify_py27 ensurepip
+  else
+    install_package "pypy-5.8-linux_x86_64-portable" "https://bitbucket.org/squeaky/portable-pypy/downloads/pypy-5.8.0-linux_x86_64-portable.tar.bz2#337fd7e947a74cb09253f5ff330fb1be471b1d7a64c2d0340ac387920e71bf3c" "pypy" verify_py27 ensurepip
```
Comment (yyuu) [inline plugins/python-build/share/python-build/pypy2.7-5.8.0]: It seems that download URL is https://bitbucket.org/squeaky/portable-pypy/downloads/pypy-5.8-linux_x86_64-portable.tar.bz2 , not pypy-5.8<strong>.0</strong>-linux_x86_64-portable.tar.bz2

File: plugins/python-build/share/python-build/pypy3.5-5.8.0 [hunk 2]
```diff
@@ -0,0 +1,18 @@
+case "$(pypy_architecture 2>/dev/null || true)" in
+"linux64" )
+  if require_distro "Ubuntu 14.04" 1>/dev/null 2>&1; then
+    install_package "pypy3-v5.8.0-linux64" "https://bitbucket.org/pypy/pypy/downloads/pypy3-v5.8.0-linux64.tar.bz2#57d871a7f1135719c138cee4e3533c3275d682a76a40ff668e95150c65923035" "pypy" verify_py35 ensurepip
+  else
+    install_package "pypy3.5-5.8-beta-linux_x86_64-portable" "https://bitbucket.org/squeaky/portable-pypy/downloads/pypy3.5-5.8.0-beta-linux_x86_64-portable.tar.bz2#76c2c4b2846e38d2b4846d4b5f9dc0bd9fd7a9b1f04beb92975ae090ad390d75" "pypy" verify_py35 ensurepip
```
Comment (yyuu) [inline plugins/python-build/share/python-build/pypy3.5-5.8.0]: It seems that the download URL need to be fixed as https://bitbucket.org/squeaky/portable-pypy/downloads/pypy3.5-5.8-beta-linux_x86_64-portable.tar.bz2

**Comment by Samureus** [comment]
Done, thanks for spotting these.

**Pull request merged**